toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt

```diff
@@ -413,6 +398,91 @@ private fun MapViewEventHandler(
     }
 }
 
+/**
+ * Handles viewpoint change events and persistence for a [MapView].
+ *
+ * @since 200.4.0
+ */
+@Composable
+private fun ViewpointHandler(
+    mapView: MapView,
+    viewpointPersistence: ViewpointPersistence,
+    onViewpointChangedForCenterAndScale: ((Viewpoint) -> Unit)?,
+    onViewpointChangedForBoundingGeometry: ((Viewpoint) -> Unit)?,
+    onVisibleAreaChanged: ((Polygon) -> Unit)?
+) {
+    val currentOnViewpointChangedForCenterAndScale by rememberUpdatedState(
+        onViewpointChangedForCenterAndScale
+    )
+    val currentOnViewpointChangedForBoundingGeometry by rememberUpdatedState(
+        onViewpointChangedForBoundingGeometry
+    )
+    val currentVisibleAreaChanged by rememberUpdatedState(onVisibleAreaChanged)
+    val currentViewpointPersistence by rememberUpdatedState(viewpointPersistence)
+    var persistedViewpoint by rememberSaveable(
+        saver = Saver(
+            save = {
+                it.value?.toJson() ?: "null"
+            },
+            restore = {
+                mutableStateOf(Viewpoint.fromJsonOrNull(it))
+            }
+        )
+    ) {
+        mutableStateOf<Viewpoint?>(null)
+    }
+
+    LaunchedEffect(Unit) {
+        persistedViewpoint?.let { mapView.setViewpoint(it) }
+        launch {
+            mapView.viewpointChanged.collect {
+                currentVisibleAreaChanged?.invoke(
+                    mapView.visibleArea
+                        ?: throw IllegalStateException("MapView visible area should not be null")
+                )
+                var currentViewpoint =
+                    if (currentViewpointPersistence != ViewpointPersistence.None) {
+                        val viewpointType =
+                            if (currentViewpointPersistence == ViewpointPersistence.ByCenterAndScale) ViewpointType.CenterAndScale else ViewpointType.BoundingGeometry
+                        mapView.getCurrentViewpoint(viewpointType)
+                    } else null
```

Comment:
Would a when statement make this more readable?
```suggestion
                var currentViewpoint = when (currentViewpointPersistence) {
                   is ViewpointPersistence.ByCenterAndScale -> mapView.getCurrentViewpoint(ViewpointType.CenterAndScale)
                   is ViewpointType.BoundingGeometry -> mapView.getCurrentViewpoint(ViewpointType.BoundingGeometry)
                   else -> null
                }
```